extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala

```diff
@@ -347,8 +347,15 @@ object PrivilegesBuilder {
         buildQuery(getQuery, inputObjs)
 
       case "CreateFunctionCommand" |
-          "DropFunctionCommand" |
-          "RefreshFunctionCommand" =>
+          "DropFunctionCommand" =>
+        val db = getPlanField[Option[String]]("databaseName")
+        val functionName = getPlanField[String]("functionName")
+        val isTemp = getPlanField[Boolean]("isTemp")
+        if (!isTemp) {
+          outputObjs += functionPrivileges(db.orNull, functionName)
+        }
```

Comment:
```suggestion
        val isTemp = getPlanField[Boolean]("isTemp")
        if (!isTemp) {
          val db = getPlanField[Option[String]]("databaseName")
          val functionName = getPlanField[String]("functionName")
          outputObjs += functionPrivileges(db.orNull, functionName)
        }
```
We could check if it's temp function first, before fetching names for db and function.